back/engines/commercial/insights/app/controllers/insights/web_api/v1/text_network_analysis_tasks_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Insights
+  module WebApi
+    module V1
+      class TextNetworkAnalysisTasksController < ::ApplicationController
+        skip_after_action :verify_policy_scoped, only: [:index]
+        after_action :verify_authorized, only: [:index]
+
+        def index
+          render json: TextNetworkAnalysisTaskViewSerializer.new(task_views, params: fastjson_params)
+        end
+
+        def task_views
+          @task_views ||= view.tna_tasks_views
+        end
+
+        # @return [Insights::View]
+        def view
+          @view ||= authorize(
+            View.includes(:scope).find(params.require(:view_id)),
+            :destroy?
```

Comment:
That's a bit weird, we won't destroy anything, we might want to set up a different policy for processing ?